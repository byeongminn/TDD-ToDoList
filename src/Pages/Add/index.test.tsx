/*
  <Add 페이지 컴포넌트 사양>
  1. 할 일을 입력할 수 있는 입력창이 있다. 할 일 입력창은 기본적으로 '할 일을 입력해 주세요.'라는 문구를 표시한다.
  2. 할 일을 추가할 수 있는 추가 버튼이 있다.
  3. 할 일을 입력하고, 추가 버튼을 누르면 할 일을 localStorage에 저장하고 할 일 목록 페이지로 이동한다.
  4. 아무 할 일이 없는 경우에는 할 일 추가 버튼을 클릭하여도 빈 할 일이 저장되지 않으며 할 일 목록 페이지로 이동하지 않는다.
*/

import { render, screen } from '@testing-library/react';
import { Add } from '.';

describe('<Add />', () => {
  it('renders component correctly', () => {
    render(<Add />);

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요.');
    expect(input).toBeInTheDocument();

    const button = screen.getByText('추가');
    expect(button).toBeInTheDocument();
  });
});
