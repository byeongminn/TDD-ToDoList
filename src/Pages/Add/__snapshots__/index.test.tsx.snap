// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Add /> renders component correctly 1`] = `
<div>
  <div
    class="sc-iGgVNO cJmkL"
  >
    <input
      class="sc-gsFSjX KFngo"
      placeholder="할 일을 입력해 주세요."
    />
    <div
      class="sc-fqkwJk kFOFCi"
    >
      <div
        class="sc-dcJtft drInyG"
      >
        추가
      </div>
    </div>
  </div>
</div>
`;
